portfolio/src/main/webapp/script.js

```diff
@@ -31,23 +31,186 @@ function addRandomFunFact() {
         unusedFunFacts[randomIndex] = firstFunFact;
         unusedFunFacts.shift();
 
-        const funFactRowText = '<tr><td>' + funfact + '</td></tr>';
-        tableText += '<tr><td>' + funfact + '</td></tr>';
-        countTableEntries++;
+        factTableText += '<tr><td>' + funfact + '</td></tr>';
         document.getElementById('funfact-message').innerHTML = "";
         
-        funFactTable.innerHTML = tableText + '</table>';
+        funFactTable.innerHTML = factTableText + '</table>';
 
   } else {
       document.getElementById('funfact-message').innerHTML = "Congrats - you've gone through all of the fun facts!";
   }
 
 }
 
+/**
+ * Fetches the comments from the server and adds it to the DOM.
+ Based on week-3-server/random-quotes/src/webapp/script.js from the Week-3-Server tutorial
+ */
+
+var shouldDisplay = true;
+function getComments(myLocation) {
+
+  //first, reset by updating num comments, which also hides comments
+  refreshNumComments(myLocation);
+  //getNumComments(myLocation);
+  
+  //toggle between hide and display
+  if (shouldDisplay) {
+
+    console.log('Fetching comments.');
+    document.getElementById(myLocation + "-get-button").innerHTML = "Hide Comments";
+
+    setLocation(myLocation);
+
+    // The fetch() function returns a Promise because the request is asynchronous.
+    const responsePromise = fetch('/comments?location=' + myLocation, {
+        location: myLocation
+    });
+
+    // When the request is complete, pass the response into handleResponse().
+    responsePromise.then(handleResponse);
+
+    shouldDisplay = false;
```

Comment:
Can you add comment on why do we toggle here?

Reply:
Add a comment saying that since comments are now shown, the button that will be displayed is "hide comments". Therefore showingDisplayCommentButton is now toggled to False.